src/Drupal/Composer/DrupalInstallerPlugin.php

```diff
@@ -622,7 +624,7 @@ protected function getPackageVersion(PackageInterface $package) {
                 }
                 else {
                     // Drupal contrib versions have three numbers, i.e. 7.x-1.7.
-                    $version = $matches[1] . '.x-' . $matches[2] . '.' . $matches[3];
+                    $version = '7.x-' . $matches[1] . '.' . $matches[2];
```

Comment:
I am actually not sure, but I found no good way to see the actual channel used for the package. Also I guess 8.x packages might get semver at some point in the future, so maybe its less important there.